synapse/handlers/federation.py

```diff
@@ -1817,14 +1817,16 @@ async def _sync_partial_state_room(
 
                 logger.info("Clearing partial-state flag for %s", room_id)
                 success = await self.store.clear_partial_state_room(room_id)
+
+                # Poke the notifier so that other workers see the write to
+                # the un-partial-stated rooms stream.
+                self._notifier.notify_replication()
+
```

Comment:
Now that we always notify replication when we're done with a stream id, is this still necessary?
I'm happy to leave it in since the performance impact should be small.

Reply:
I'll merge this for now, and let @erikjohnston come back to this when he's back. There were one or two test failures, but none on a retry. So I guess we have some flakes to deal with.